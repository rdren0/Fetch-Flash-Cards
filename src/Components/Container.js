import React, { Component } from 'react';
import Card from './Card';
import PlaceHolder from './PlaceHolder';

class Container extends Component {
  constructor(props) {
    super(props);

  }

  render() {
    let completionStatus;
    let cards;
    if(this.props.data.length){
    cards = this.props.data.map(card =>{
      if(this.props.completed.includes(card.ID)){
        completionStatus =true;
      }else{
        completionStatus =false; 
      }

    return <Card
      ID={card.ID}
      Question = {card.Question}
      Answer = {card.Answer}
      Difficulty = {card.Difficulty}
      completed = {this.props.completed}
      addCompleted = {this.props.addCompleted}
      status = {completionStatus}
              />
      }); 
    }else{
      cards = <PlaceHolder />
    }

    return (
      <div  className="container" >
        {cards}
      </div>
    )
  }  
}
      // cards = <iframe src="https://giphy.com/embed/xlYKItjhiDsY" width="480" height="270" frameBorder="0" class="giphy-embed" allowFullScreen></iframe>


export default Container;


      