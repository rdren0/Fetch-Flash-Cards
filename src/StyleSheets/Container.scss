.container{
  background-color: #2E6171;
}