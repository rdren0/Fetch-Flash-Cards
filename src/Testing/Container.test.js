import Container from '../Components/Container';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let mockData = [{
  "ID": 1,
  "Question": "What status code indicates that there has been an error on the client side?",
  "Answer": 400,
  "Difficulty": "Easy"
}, {
  "ID": 2,
  "Question": "What status code indicates that there has been an error on the server side?",
  "Answer": 500,
  "Difficulty": "Easy"
}, {
  "ID": 3,
  "Question": "What status code indicates the fetch was successful but the data may have moved?",
  "Answer": 300,
  "Difficulty": "Easy"
}]

let mockData2 =[]

let mockComplete = [2,3]

describe('Container', () => {
  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallow(
      <Container
      data = {mockData}
      completed = {mockComplete}/>
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a placeholder if data is empty', () => {
     <Container
      data = {mockData2}/>
  });

  it('should render a Fetch Component if all cards completed', () => {
     <Container
      completed = {30}/>
  });
});