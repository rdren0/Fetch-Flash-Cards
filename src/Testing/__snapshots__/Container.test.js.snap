// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot with all data passed in correctly 1`] = `
<div
  className="container"
>
  <Card
    Answer={400}
    Difficulty="Easy"
    ID={1}
    Question="What status code indicates that there has been an error on the client side?"
    completed={
      Array [
        2,
        3,
      ]
    }
    key="1"
    status={false}
  />
  <Card
    Answer={500}
    Difficulty="Easy"
    ID={2}
    Question="What status code indicates that there has been an error on the server side?"
    completed={
      Array [
        2,
        3,
      ]
    }
    key="2"
    status={true}
  />
  <Card
    Answer={300}
    Difficulty="Easy"
    ID={3}
    Question="What status code indicates the fetch was successful but the data may have moved?"
    completed={
      Array [
        2,
        3,
      ]
    }
    key="3"
    status={true}
  />
</div>
`;
